pkg/chunk/storage/caching_index_client.go

```diff
@@ -115,12 +123,17 @@ func (s *cachingIndexClient) QueryPages(ctx context.Context, queries []chunk.Ind
 	)
 
 	for _, key := range misses {
-		// Only need to consider one of the queries as they have the same table & hash.
 		queries := queriesByKey[key]
-		cacheableMissed = append(cacheableMissed, chunk.IndexQuery{
-			TableName: queries[0].TableName,
-			HashValue: queries[0].HashValue,
-		})
+		if isChunksQuery(queries[0]) {
+			// Only need to consider one of the queries as they are the same.
+			cacheableMissed = append(cacheableMissed, chunk.IndexQuery{
+				TableName: queries[0].TableName,
+				HashValue: queries[0].HashValue,
+			})
+		} else {
+			// We would always get distinct queries, if not then something is wrong and consider just one of it.
```

Comment:
I don't understand "and consider just one of it".
Are you saying that the keys created on line 93 should be unique, therefore `queries` here should only have one value?
If so, should that assumption be checked on line 95?

Reply:
yeah, you are right, the check should be moved to 95 instead. Will take care of it, thanks!